src/darwin/Framework/CHIP/templates/CHIPClustersObjc-src.zapt

```diff
@@ -174,10 +144,40 @@ private:
     return self;
 }
 
+- (void)dispatchAction:(CHIPCallbackBridge *)callbackBridge withBlock:(CHIPActionBlock)action
+{
+    callbackBridge->SetDispatchQueue([self callbackQueue]);
+
+    dispatch_sync([self chipWorkQueue], ^{
+        CHIP_ERROR err = action();
+        if (CHIP_NO_ERROR != err)
+        {
+            dispatch_async([self callbackQueue], ^{
+              callbackBridge->GetResponseHandler()([CHIPError errorForCHIPErrorCode:err], nil);
+              delete callbackBridge;
+            });
+        }
+    });
+}
+
 - (Controller::ClusterBase *)getCluster
 {
     return nullptr;
 }
+
+- (chip::ByteSpan)asSpan:(id)value
+{
+  if ([value isKindOfClass:[NSData class]])
+  {
+      NSData * v = (NSData *) value;
+      return chip::ByteSpan((const uint8_t*)v.bytes, v.length);
+  }
+  else
+  {
+      NSString * v = (NSString *) value;
+      return chip::ByteSpan((const uint8_t*)[v dataUsingEncoding:NSUTF8StringEncoding].bytes, [v lengthOfBytesUsingEncoding:NSUTF8StringEncoding]);
```

Comment:
Does the return value from `dataUsingEncoding` stay alive after this function returns?  We're returning a ByteSpan pointing into it.....

I would not have expected it to survive, myself.

Reply:
My understanding is yes. The value returned by `dataUsingEncoding` is an `autoreleased` value that will be drained once the event loop has pass is over.
The current code does use the `ByteSpan` into a `dispatch_sync` call to the stack, and I would expect the underlying data to still be alive.